src/Commands/RebuildOaiEntries.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Drupal\islandora_drush_utils\Commands;
+
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+use Drupal\Core\Queue\QueueFactory;
+use Drush\Commands\DrushCommands;
+
+/**
+ * RebuildOaiEntries commands.
+ *
+ * These commands rebuild the OAI entries and consume them.
+ */
+class RebuildOaiEntries extends DrushCommands {
+
+  use DependencySerializationTrait;
+
+  /**
+   * The queue service.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queue;
+
+  /**
+   * Set variables to used across commands.
+   *
+   * @param \Drupal\Core\Queue\QueueFactory $queue
+   *   The queue service.
+   */
+  public function __construct(QueueFactory $queue) {
+    $this->queue = $queue;
+  }
+
+  /**
+   * Rebuild OAI entries.
+   *
+   * @param array $options
+   *   Additional command options.
+   *
+   * @option batchSize An integer of the amount of entries to process per batch.
+   *
+   * @command islandora_drush_utils:rebuild-oai
+   * @aliases idr:roai
+   */
+  public function rebuild(array $options = [
+    'batchSize' => 10,
+  ]) {
+    rest_oai_pmh_cache_views();
+
+    $batch = [
+      'title' => 'Processing OAI rebuild',
+      'finished' => 'rest_oai_pmh_batch_finished',
+      'operations' => [
+        [
+          [$this, 'rebuildBatch'],
+          [$options['batchSize']],
+        ],
+      ],
+    ];
+
+    batch_set($batch);
+    drush_backend_batch_process();
+  }
+
+  /**
+   * Batch for processing OAI queue.
+   *
+   * @param int $batch_size
+   *   The number of nodes to process at a single time.
+   * @param array|\DrushBatchContext $context
+   *   The batch context.
+   */
+  public function rebuildBatch(int $batch_size, &$context) {
+    $queue = $this->queue->get('rest_oai_pmh_views_cache_cron');
+
+    // Setting the defaults.
+    if (empty($context['sandbox'])) {
+      $context['sandbox']['processed_items'] = 0;
+      $context['sandbox']['total_items'] = $queue->numberOfItems();
+    }
+
+    $item = $queue->claimItem();
+    if (!$item) {
+      // Queue exhausted; we're done.
+      $context['finished'] = 1;
+      return;
+    }
+
+    // Process queue items.
+    rest_oai_pmh_process_queue($item);
+    $context['sandbox']['processed_items']++;
+
+    // Set the batch progress percentage.
+    $context['finished'] = $context['sandbox']['processed_items'] / $context['sandbox']['total_items'];
```

Comment:
Probably fine, but yeah, with the suggested change from https://github.com/discoverygarden/islandora_drush_utils/pull/25#discussion_r1592552267, not allowing this to be `1` here was intentional, so it _always_ terminate the batch via the `return` above ( https://github.com/discoverygarden/islandora_drush_utils/pull/25/files#diff-92200b5dd6599050e6d25fab166a086f39fa70bc862adf6b6e326683911562f6R84-R88 )

Otherwise, could be unexpected race-conditions where on-going interactions lead to new items in the queue that were not accounted for in the original `numberOfItems()` count, and so potentially leaving entries in the queue when our `processed_items` value becomes greater than the original value. Would probably be rare, and the usual cron-running should pick up anything that were skipped on its next pass.